### ISSUE TITLE
(documentation flaw) JSXGraph: < and > are rewritten as HTML character codes

### ISSUE THREAD

**Issue by hgeorgsch** [maths/moodle-qtype_stack#816]
I am aware of the problems with the Moodle HTML editor, which causes a lot of trouble, but now I run into a problem even using the plain text editor for question text.

As a minimal example, we try the following loop in question text.

```
[[jsxgraph height='850px' width='850px']]
for (var i = 0; i<5; i++) {
   console.log(i);
}
[[/jsxgraph]]
```

The less than relation is rewritten as `&lt;` upon save, and the js interpreter reports syntax error on the unexpected semicolon.

I tried to browse the documentation without finding anything on this.  It seems that all the examples avoid less than/greater than completely.

1.  Is there a work-around that I have missed?
2.  Is it possible to add a filter into STACK to undo the character rewrite?

**Comment by aharjula** [comment]
1. Maybe try writing as something not requiring those chars? JSXGraph itself provides [functions](https://jsxgraph.org/docs/symbols/JXG.Math.html#.lt) for writing these comparisons for people that use other tools. Personally, I might write `i!=5` in that particular loop.

2. No, it is impossible to know when the entity is meant to be there and when not. It is just bad tooling that breaks input and needs to be fixed elsewhere.

In any case, for me in a 4.4 test installation with Moodle 4 (also in 3.11) default settings, everything just worked. So maybe there is something else affecting things, or maybe this "BUG" has already been fixed, although I doubt that as the whole idea of having our own `[[jsxgraph]]` block is to be able to act on things before Moodle filters get a chance to do anything and input has not been a problem with the plain text editor in the past.

**Comment by hgeorgsch** [comment]
Thanks a lot.

May I suggest adding a link to the rewriting functions you mention from the [docs](https://stack-demo.maths.ed.ac.uk/demo/question/type/stack/doc/doc.php/Authoring/JSXGraph.md) before you close the issue?

If the bug was fixed in 3.11, it is great.  I was testing on 3.9 LTS.  If STACK had been released for Moodle 4 I would have focused on that.

**Comment by sangwinc** [comment]
Sorry this caused you problems.  I'd very much welcome improvements in the docs.  Could you make a very specific suggestion, a pull request even, and I'll add this to the docs?
